robot :
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # gazebo_ros_control:
  #   type: effort_controllers/JointTrajectoryController
  #   joints:
  #       - joint1
  #       - joint2
  #       - joint3

  #   pid_gains:
  #       joint1: { p: 12000, d: 50, i: 0.0, i_clamp: 10000 }
  #       joint2: { p: 30000, d: 100, i: 0.02, i_clamp: 10000 }
  #       joint3: { p: 18000, d: 50, i: 0.01, i_clamp: 1 }

  #   constraints:
  #       goal_time: 2.0

  #   state_publish_rate: 25
  
  # Position Controllers ------------------------------------
  joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint1
    # pid: {p: 25.0, i: 0.0, d: 10}
    pid: {p: 150.0, i: 0.00, d: 80.0}

  joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint2
    # pid: {p: 10.0, i: 0.00, d: 1.0}
    pid: {p: 150.0, i: 0.00, d: 80.0}
  
  joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint3
    # pid: { p: 75, i: 20.00, d: 40.0 }
    # pid: { p: 20, i: 0.00, d: 10, i_clamp: 1 }
    pid: { p: 10, i: 0.00, d: 0.3, i_clamp: 1 }



  


